govcd/disk.go

```diff
@@ -350,6 +350,31 @@ func (vdc *Vdc) QueryDisk(diskName string) (DiskRecord, error) {
 	return *newDisk, nil
 }
 
+// Find independent disks using disk name. Returns list of VMRecord query type
+func (vdc *Vdc) QueryDisks(diskName string) (*[]*types.DiskRecordType, error) {
+
+	if diskName == "" {
+		return nil, fmt.Errorf("disk name can not be empty")
+	}
+
+	typeMedia := "disk"
+	if vdc.client.IsSysAdmin {
+		typeMedia = "adminDisk"
+	}
+
+	results, err := vdc.QueryWithNotEncodedParams(nil, map[string]string{"type": typeMedia, "filter": "name==" + url.QueryEscape(diskName)})
+	if err != nil {
+		return nil, fmt.Errorf("error querying disk %#v", err)
+	}
+
+	diskResults := results.Results.DiskRecord
+	if vdc.client.IsSysAdmin {
+		diskResults = results.Results.AdminDiskRecord
```

Comment:
What's the difference between `DiskRecord` and `AdminDiskRecord`? 
Does the caller need to know?

Reply:
It is how you should call and what type receive to/from Query with different type of users `system` -`none system`. I made it that way that user don't know the details.